galsim/phase_screens.py

```diff
@@ -0,0 +1,646 @@
+                if len(v) != 1:
+                    raise ValueError("Cannot broadcast kwargs of different non-length-1 lengths.")
+                out[k] = v[0]
+            except TypeError:  # Value is not list-like, so broadcast it in its entirety.
+                out[k] = v
+            except:
+                raise "Cannot broadcast kwarg {1}={2}".format(k, v)
+        yield out
+
+
+def Atmosphere(screen_size, rng=None, **kwargs):
+    """Create an atmosphere as a list of turbulent phase screens at different altitudes.  The
+    atmosphere model can then be used to simulate atmospheric PSFs.
+
+    Simulating an atmospheric PSF is typically accomplished by first representing the 3-dimensional
+    turbulence in the atmosphere as a series of discrete 2-dimensional phase screens.  These screens
+    may blow around in the wind, and may or may not also evolve in time.  This function allows one
+    to quickly assemble a list of atmospheric phase screens into a galsim.PhaseScreenList object,
+    which can then be used to evaluate PSFs through various columns of atmosphere at different field
+    angles.
+
+    The atmospheric screens currently available both produce turbulence that follows a von Karman
+    power spectrum.  Specifically, the phase power spectrum in each screen can be written
+
+    psi(nu) = 0.023 r0^(-5/3) (nu^2 + 1/L0^2)^(11/6)
+
+    where psi(nu) is the power spectral density at spatial frequency nu, r0 is the Fried parameter
+    (which has dimensions of length) and sets the amplitude of the turbulence, and L0 is the outer
+    scale (also dimensions of length) beyond which the power asymptotically flattens.  Typical
+    values for r0 are ~0.1 to 0.2 meters, which corresponds roughly to PSF FWHMs of ~0.5 to 1.0
+    arcsec for optical wavelengths.  Note that r0 is a function of wavelength, scaling like
+    r0 ~ wavelength^(6/5).  To reduce confusion, the input parameter here is named r0_500 and refers
+    explicitly to the Fried parameter at a wavelength of 500 nm.  The outer scale is typically in
+    the 10s of meters and does not vary with wavelength.
+
+    To create multiple layers, simply specify keyword arguments as length-N lists instead of scalars
+    (works for all arguments except `time_step` and `rng`).  If, for any of these keyword arguments,
+    you want to use the same value for each layer, then you can just specify the argument as a
+    scalar and the function will automatically broadcast it into a list with length equal to the
+    longest found keyword argument list.  Note that it is an error to specify keywords with lists of
+    different lengths (unless only one of them has length > 1).
+
+    The one exception to the above is the keyword `r0_500`.  The effective Fried parameter for a set
+    of atmospheric layers is r0_500_effective = (sum(r**(-5./3) for r in r0_500s))**(-3./5).
+    Providing `r0_500` as a scalar or length-1 list will result in broadcasting such that the
```

Comment:
I originally misread this as "length minus one" list.  Maybe "single-element list" instead.